
## Monitoring / documenting the bioinformatic QC on Genomic Status

Open [GenStat deliveries](<genstat_url>/deliveries) and find the project you are interested in, and click on the project id to open the project status page.

::: {.callout-note}
If you provided the project id when generating this checklist, you can access it via [this link](<genstat_url>/project/<project_id>).
:::

- [ ] Check the running notes for anything that needs to be taken into account during the QC

## Setting values of QC boxes

Click on the [bioinfo tab](<genstat_url>/bioinfo/<project_id>) and the [flowcell page](<genstat_url>/flowcells/<flowcell_id>)

On the flowcell, check the following:

- [ ] Total clusters and clusters per lane (see @tbl-yields)
- [ ] PhiX error rate -> It should be below 2% (unless more than 1% PhiX was added)
- [ ] Q30 (see @tbl-q30)
- [ ] Undetermined reads
  - [ ] Total number is below threshold (see @tbl-undetermined)
  - [ ] The majority belong to the same index (e.g. PhiX)
- [ ] The fraction number of any unexpected indexes and if they are similar to any index in the project

On the bioinfo page:

- [ ] Set the status according to the results above
- [ ] Click save

On the project page:

- [ ] Check the number of lanes ordered and obtained

## Documenting the status in the Clarity LIMS

Open the LIMS _BCL conversion_ page:

- On GenStat, go to the _Samples_ tab
- Click on any of the samples in the _Sample_ column (the first time to open the LIMS login page, and the second time to open the samples view)
- Click on the _BCL conversion_ button

On the LIMS page:

::: {.callout-important}
Although possible, it is better to process one sample at the time
:::

- [ ] In the dropdown menu, change from _Container_ to _Project_
- [ ] Click on _add group_ for the samples to process
- [ ] Click on _Begin work_ at the top right corner

In the next section:

- [ ] Update the Document version
- [ ] Click on _BCL conversion and demultiplexing_

Once the import is complete:

- [ ] Update the _include reads_ column
  - [ ] From the dropdown menu, select _Include reads_
  - [ ] From the second dropdown menu, select _Yes_
  - [ ] Click on _Apply_

- [ ] Double-check that the _Include reads_ column is set to _Yes_ for all samples
- [ ] Click on _Next step_ at the top right corner

On the next page:

- [ ] Select _Mark protocol as complete_ in the dropdown menu
- [ ] Click on _Finish step_ at the top right corner


Re-open the LIMS _Project summary_ page:

- On GenStat, go to the _Samples_ tab
- Click on any of the samples in the _Sample_ column (the first time to open the LIMS login page, and the second time to open the samples view)
- Click on the _Project Summary_ button

On the LIMS page:

::: {.callout-warning}
This action should be performed only if no more samples are expected to be sequenced
:::

- [ ] Set _All samples as sequenced_ by writing your signature
- [ ] Check other fields for correctness
- [ ] Click on _Reads Aggregation | Run_

In the table at the bottom of the page, the _QC_ column refers to whether the sample has to be delivered or not.

- [ ] Set the value of the _QC_ column accordingly
- [ ] Set the value of the _Set App QC_ column to _N/A_ (unless they are RNA-seq samples)
- [ ] Click on _Record Changes | Run_

### Data organisation on the UPPMAX cluster

- [ ] Log into Miarka
- [ ] Start the data organisation by running the following command:
```bash
ngi_pipeline_start.py organize flowcell <flowcell_id> -p <project_id>
```
::: {.callout-tip}
The `project_id` and `flowcell_id` can be found either in the GenStat running notes or at the top of the bioinfo tab
:::

Once the data organisation is complete:

- [ ] Check on [Charon](<charon_url>/project/<project_id>) that the _Status_ column changed from _FRESH_ to _STALE_

### Basic QC reports and contamination screen

On Miarka:

- [ ] Submit the QC jobs as follows:

```bash
ngi_pipeline_start.py qc project <ngi_path>/DATA/<project_id>
```

- [ ] Write a running note that the fastqc and fast screen are done/started

```text
Data organized and QC started for <project_id> samples on <flowcell_id>.
```

- [ ] Sign the note and add the required tags

You can check that the jobs have been submitted or are running by using the following command:
```bash
jobinfo | grep <project_id>
```

### Contamination screen / MultiQC report generation

Once the jobs are finished (i.e. the `jobinfo` command does not return any entry):

- [ ] Move to the project directory and submit the multiqc job:
```bash
cd <ngi_path>/ANALYSIS/<project_id> && multiqc qc_ngi
```

In the command line output:

- [ ] Check that the total number of reports matches the number of samples (it should be twice the number of samples)

- [ ] If opened, refresh the GenStat page, so that the MultiQC report will be listed in the _Reports_ tab

Open the MultiQC report and check the following:

  - [ ] The FastQ Screen results match the expectations, based on the information about the samples
  - [ ] The duplication levels are within acceptable limits
  - [ ] Mean Quality Scores are in the green zone

Open the [bioinfo tab](<genstat_url>/bioinfo/<project_id>):

- [ ] Fill all the fields according to the results of the MultiQC report

::: {.callout-tip title="Write a running note [OPTIONAL]"}
- [ ] Write a running note if there are things to bring up from the report: `MultiQC report has been generated for <project_id>.`
- [ ] Sign the note and add the required tags
:::

---

## Appendix

### Positive and negative controls

Positive and negative controls are valid for a calendar year, and can be found in their own P.Ositive and N.Egative projects. In general, add them only if it explicitly stated in the running notes.

### Lane yields

| Platform | Type | Yield (Millions) |
| :---: | :---: | :---: |
| MiSeq | v2 | 10 |
| MiSeq | v3 | 18 |
| MiSeq | Nano v2 | 0.75 |
| NextSeq 2000 | P1 | 100 |
| NextSeq 2000 | P2 | 400 |
| NextSeq 2000 | P3 | 1100 |
| NovaSeq 6000 | SP | 325 |
| NovaSeq 6000 | S1 | 650 |
| NovaSeq 6000 | S2 | 1650 |
| NovaSeq 6000 | S4 | 2000 |
| NovaSeqXPlus | 1.5B | 750 |
| NovaSeqXPlus | 10B | 1000 |
| NovaSeqXPlus | 25B | 3000 |

: The reads are distributed over two lanes that are not physically separated but listed as separate lanes in the output file structure. {#tbl-yields .striped .hover}

### Q30 thresholds

| | MiSeq | NextSeq | NovaSeq |
| :---: | :---: | :---: | :---: |
| 2x50 bp (or below) | 80% | 80% | 85% |
| 2x100 bp | 75% | 80% | 80% |
| 2x150 bp | 70% | 75% | 75% |
| 2x250 bp | 60% | 75% | 75% |

: Base-calling accuracy based on the Q score (Phread quality score) of the reads. {#tbl-q30 .striped .hover}

### Undetermined reads thresholds

| Platform | Setup | WARN threshold | FAIL threshold |
| :---: | :---: | :---: | :---: |
| MiSeq| Single index read | 10% | 20% |
| MiSeq| Dual index read | 20% | 30% |
| NextSeq| Any | 10% | 20% |
| NovaSeq| Any | 10% | 20% |

: The thresholds are based on amount of PhiX added to the library (i.e. 1%); if more than 1% is added, the thresholds should be adjusted accordingly. {#tbl-undetermined .striped .hover}

---
